datadog/dogstatsd/container.py

```diff
@@ -55,3 +76,30 @@ def _read_container_id(self, fpath):
         except Exception as e:
             raise UnresolvableContainerID("Unable to read the container ID: " + str(e))
         return None
+
+    def _get_cgroup_from_inode(self):
+        """Read the container ID from the cgroup inode."""
+        # Parse /proc/self/cgroup and get a map of controller to its associated cgroup node path.
+        cgroup_controllers_paths = {}
+        with open(self.CGROUP_PATH, mode="r") as fp:
+            for line in fp:
+                tokens = line.strip().split(":")
+                if len(tokens) != 3:
+                    continue
+                if tokens[1] == self.CGROUPV1_BASE_CONTROLLER or tokens[1] == "":
+                    cgroup_controllers_paths[tokens[1]] = tokens[2]
+
+        # Retrieve the cgroup inode from "/sys/fs/cgroup + controller + cgroupNodePath"
+        for controller in [
+            self.CGROUPV1_BASE_CONTROLLER,
+            self.CGROUPV2_BASE_CONTROLLER,
+        ]:
+            if controller in cgroup_controllers_paths:
+                inode_path = os.path.join(
+                    self.DEFAULT_CGROUP_MOUNT_PATH,
+                    controller,
+                    cgroup_controllers_paths[controller] if cgroup_controllers_paths[controller] != "/" else "",
+                )
+                return "in-{0}".format(os.stat(inode_path).st_ino)
```

Comment:
Does this need to check the inode number? https://github.com/DataDog/java-dogstatsd-client/pull/232/files#diff-79768f0f7e1c9370c006cfd086b4ae68cdc0995de5cf385a40d2d5312dddecbbR176

Reply:
Yes it would be better, I've added a commit for this 👍